Lesson18/serkan/script.js

```diff
@@ -0,0 +1,182 @@
+// 1. Write a function that takes an array of integers and returns the sum of all elements.
+// Input: [1, 2, 3, 4]
+// Output: 10
+const integersEx1 = [1, 2, 3, 4];
+const integersEx1V2 = [10, 2, 30, 4];
+
+function sumofArray (integersEx1) {
+let sum = 0;
+for (let i=0; i < integersEx1.length; i++) {
+  sum+= integersEx1[i];
+
+}
+return sum;
+}
+
+console.log(sumofArray([1, 2, 3, 4]));
+console.log("Exercise 1: ");
+console.log('----------------------------');
+// 2. Write a function that takes an array of integers and returns the average of all elements.
+//Input: [1, 2, 3, 4]
+// Output: 2.5
+
+const integersEx2 = [1, 2, 3, 4];
+function getAvgArray(array) {
+  let sum = 0;
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  const avgArray = sum / array.length;
+  return avgArray;
+}
+
+console.log(getAvgArray([1, 2, 3, 4]));
+console.log("Exercise 2:");
+
+console.log('----------------------------');
+// 3. Write a function that takes an array of strings and concatenates them together adding spaces.
+// Input: ["Hello", "World", "!"]
+// Output: "Hello World !"
+
+let stringE3 = ["Hello", "World", "!"];
+
+console.log(stringE3.join(' '));
+
+
+console.log("Exercise 3:");
+
+console.log('----------------------------');
+// 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 3, 7]
+const arrayAllNumbers = [1, 2, 3, 4, 7, 10];
+function takeOddNumber (array) {
+  let oddNumber = [];
+  for (let i=0; i< array.length; i++) {
+    if (array[i] % 2 !== 0) {
+      oddNumber.push(array[i]);
+    }
+  }
+  return oddNumber; 
+}
+console.log(takeOddNumber(arrayAllNumbers));
+
+console.log("Ex4");
+console.log('----------------------------');
+// 5. Write a function that takes an array of integers and returns a new array with even numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 2, 4, 10]
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+function takeEvenNumber (array) {
+  let evenNumber = [];
+  for (let i=0; i< array.length; i++) {
+    if (array[i] % 2 === 0) {
+      evenNumber.push(array[i]);
+    }
+  }
+  return evenNumber; 
+}
+console.log(takeEvenNumber(testArrayEx5));
+
+// I couldn't include '1' number here, I don't know how to do it.
```

Comment:
There's a mistake in the note. The array should not include number 1 as it's an odd number.